src/core/control/tools/TextEditor.cpp

```diff
@@ -117,15 +122,20 @@ static auto cloneWithInsertToStdString(GtkTextBuffer* buf, std::string_view inse
     return res;
 }
 
-TextEditor::TextEditor(Control* control, const PageRef& page, GtkWidget* xournalWidget, double x, double y):
+TextEditor::TextEditor(Control* control, XojPageView* pageView, GtkWidget* xournalWidget, double x, double y):
         control(control),
-        page(page),
+        page(pageView->getPage()),
+        pageView(pageView),
         xournalWidget(xournalWidget),
         textWidget(gtk_xoj_int_txt_new(this), xoj::util::adopt),
         imContext(gtk_im_multicontext_new(), xoj::util::adopt),
         buffer(gtk_text_buffer_new(nullptr), xoj::util::adopt),
         viewPool(std::make_shared<xoj::util::DispatchPool<xoj::view::TextEditionView>>()) {
 
+    this->contextMenu = std::make_unique<TextEditorContextMenu>(control, this, pageView, xournalWidget);
+
+    this->pageView->getZoomControl()->addZoomListener(this);
```

Comment:
This breaks the model-view-controller structure of the code. `TextEditor` is a controller, and it should not know anything about the view.
I'd say the ContextMenu should rather be owned by the TextEditorView instance(s).